cmd/cronosd/cmd/fix-unlucky-tx.go

```diff
@@ -147,47 +151,93 @@ func FixUnluckyTxCmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			if len(blocksFile) > 0 {
-				// read block numbers from file, one number per line
-				file, err := os.Open(blocksFile)
-				if err != nil {
-					return err
-				}
-				defer file.Close()
-				scanner := bufio.NewScanner(file)
-				for scanner.Scan() {
-					blockNumber, err := strconv.ParseInt(scanner.Text(), 10, 64)
+			concurrency, err := cmd.Flags().GetInt(FlagConcurrency)
+			if err != nil {
+				return err
+			}
+
+			blockChan := make(chan int64, concurrency)
+			var wg sync.WaitGroup
+			ctCtx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			for i := 0; i < concurrency; i++ {
+				wg.Add(1)
+				go func(wg *sync.WaitGroup) {
+					defer wg.Done()
+					for {
+						select {
+						case <-ctCtx.Done():
```

Comment:
should we select the `Done()` together with the `blockChain`?
it seems doing dead loop after finish the works.

Reply:
I guess we need to call the cancel no mater if `findBlock` returns error, right?